### PR TITLE
[libc++] Fixed uniform_real_distribution.h where it was allowing initialization with non floating point types

### PR DESCRIPTION
**Pull request by Pranav10903**
Included a static_assert to deal with cases where result_types are initialized with non floating point type values which can cause undefined behaviour as mentioned [here](https://en.cppreference.com/w/cpp/numeric/random/uniform_real_distribution)

### CODE REVIEW

File: libcxx/include/__random/uniform_real_distribution.h [hunk 1]
```diff
@@ -14,6 +14,7 @@
 #include <__random/is_valid.h>
 #include <iosfwd>
 #include <limits>
+#include <type_traits>
```
Comment (philnik777) [inline libcxx/include/__random/uniform_real_distribution.h:17]: Please include `<__type_traits/is_floating_point.h>` instead.

File: libcxx/include/__random/uniform_real_distribution.h [hunk 2]
```diff
@@ -27,6 +28,7 @@ _LIBCPP_BEGIN_NAMESPACE_STD
 template<class _RealType = double>
 class _LIBCPP_TEMPLATE_VIS uniform_real_distribution
 {
+     static_assert(std::is_floating_point<_RealType>::value, "result_type must be a floating point type");
```
Comment (philnik777) [inline libcxx/include/__random/uniform_real_distribution.h:31]: This needs a test. You can take a look at `libcxx/test/libcxx/input.output/iostream.format/input.streams/traits_mismatch.verify.cpp` for an example of how to check static assertions.

**Review by philnik777 — CHANGES_REQUESTED** [review]
Thanks for working on this! The code itself looks fine, but this needs a test.

**Comment by philnik777** [comment]
You should also mention the issue in the description.

**Comment by Pranav10903** [comment]
Thank you for your guidance, I will do the required changes. Thank you.

**Pull request closed without merging**